edx_solutions_api_integration/models.py

```diff
@@ -156,3 +159,187 @@ class APIUser(User):
     class Meta(object):
         """ Meta attribute to make this a proxy model"""
         proxy = True
+
+
+class PasswordHistory(models.Model):
+    """
+    This model will keep track of past passwords that a user has used
+    as well as providing contraints (e.g. can't reuse passwords)
+    """
+    class Meta(object):
+        app_label = "student"
+
+    user = models.ForeignKey(User)
+    password = models.CharField(max_length=128)
+    time_set = models.DateTimeField(default=timezone.now)
+
+    def create(self, user):
+        """
+        This will copy over the current password, if any of the configuration has been turned on
+        """
+        if not (PasswordHistory.is_student_password_reuse_restricted() or
+                PasswordHistory.is_staff_password_reuse_restricted() or
+                PasswordHistory.is_password_reset_frequency_restricted() or
+                PasswordHistory.is_staff_forced_password_reset_enabled() or
+                PasswordHistory.is_student_forced_password_reset_enabled()):
+
+            return
+
+        self.user = user
+        self.password = user.password
+        self.save()
+
+    @classmethod
+    def is_student_password_reuse_restricted(cls):
+        """
+        Returns whether the configuration which limits password reuse has been turned on
+        """
+        min_diff_pw = settings.ADVANCED_SECURITY_CONFIG.get(
+            'MIN_DIFFERENT_STUDENT_PASSWORDS_BEFORE_REUSE', 0
```

Comment:
We need to check if these variables are defined in ironwood or not? 
ADVANCED_SECURITY_CONFIG
MIN_DIFFERENT_STUDENT_PASSWORDS_BEFORE_REUSE

Reply:
These variables are not defined in ironwood we have to add it explicitly.

Reply:
@attiyaIshaque where these variables have been added?

Reply:
@ihtram In lms.env.json.